### PR TITLE
clippy: fix pedantic warnings (redundant_else) and disable rules (single_match_else, default_trait_access)

### PR DESCRIPTION
**Pull request by s373r**
## Description

Fixed clippy warnings
- `pedantic::redundant_else`

Disabled rules
- `pedantic::single_match_else`
- `pedantic::default_trait_access`

Related issue: https://github.com/kamu-data/kamu-cli/issues/332

## Checklist before requesting a review
- [x] [CHANGELOG.md](./CHANGELOG.md) updated
- [x] API changes are backwards-compatible
- [x] Workspace layout changes include a migration
- [x] Documentation update PR: <link or N/A>
- [x] Dataset pipelines update scheduled if needed

### CODE REVIEW

File: src/domain/opendatafabric/src/identity/grammar.rs [hunk 1]
```diff
@@ -172,24 +172,23 @@ impl Grammar {
...
+            return Some((repo, None, ds, tail));
+        };
+
+        let Some((acc, "")) = Self::match_account_name(acc) else {
+            return None;
+        };
+
+        let (ds, tail) = Self::match_dataset_name(rest)?;
+
+        Some((repo, Some(acc), ds, tail))
```
Comment (zaychenko-sergei) [inline src/domain/opendatafabric/src/identity/grammar.rs:192]: I don't see any positive gain from this rule. Previous structure  was easier to follow

File: src/adapter/http/src/http_server_dataset_router.rs [hunk 2]
```diff
@@ -84,19 +84,24 @@ pub async fn platform_token_validate_handler(
...
-            .unwrap(),
-        CurrentAccountSubject::Anonymous(reason) => axum::response::Response::builder()
-            .status(match reason {
-                AnonymousAccountReason::AuthenticationExpired => http::StatusCode::UNAUTHORIZED,
-                AnonymousAccountReason::AuthenticationInvalid => http::StatusCode::BAD_REQUEST,
-                AnonymousAccountReason::NoAuthenticationProvided => http::StatusCode::BAD_REQUEST,
-            })
-            .body(Default::default())
-            .unwrap(),
+    #[allow(clippy::default_trait_access)]
```
Comment (zaychenko-sergei) [inline src/adapter/http/src/http_server_dataset_router.rs:87]: This rule seems pointless

**Review by zaychenko-sergei — COMMENTED** [review]
This rule might be somewhat useful: `redundant_else = "warn"`

Other two do not seem to add any value. 
   `single_match_else = "warn"`
   `default_trait_access = "warn"`

I don't think that a nested matches are harder to read than multiple if-lets with returns. 
`Default::default()` is a Rust idiomatic way, and type can be visible just by pointing the mouse in IDE.

File: src/adapter/http/src/http_server_dataset_router.rs (hunk 2 above)
Comment (s373r) [reply src/adapter/http/src/http_server_dataset_router.rs]: The rule was disabled

File: src/domain/opendatafabric/src/identity/grammar.rs (hunk 1 above)
Comment (s373r) [reply src/domain/opendatafabric/src/identity/grammar.rs]: The rule was disabled; these changes were reverted

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**